server/src/testFixtures/java/org/elasticsearch/test/IntegTestCase.java

```diff
@@ -311,7 +312,10 @@ public abstract class IntegTestCase extends ESTestCase {
     private static Long SUITE_SEED = null;
 
     @Rule
-    public Timeout globalTimeout = new Timeout(5, TimeUnit.MINUTES);
+    public Timeout globalTimeout = ManagementFactory.getRuntimeMXBean().getInputArguments().stream()
+        .anyMatch(s -> s.contains("-agentlib:jdwp"))
+            ? new Timeout(30, TimeUnit.MINUTES)
+            : new Timeout(3, TimeUnit.MINUTES);
```

Comment:
just curious: any reason to lower it?

Reply:
In https://github.com/crate/crate/pull/16964 I made some changes that led to lots of test timeouts and if each suite is stuck for 5 minutes it takes ages to get the results and see which tests are having issues.

I even lowered it to 20sec on the branch to get results within a reasonable amount of time but based on the slow test case durations on https://wacklig.pipifein.dev/github/crate/crate and accounting for slower github actions that would be too low.

I'm thinking ideally we'd set a lower default and override in test suites as needed. That would also help us get a bit more conscious about test speed (and regressions in that regard) but I didn't want to open that box right now.

Reply:
We should probably also do the same jdwp trick for the statement timeout used on `execute`. Avoids running into timeouts when you're too slow stepping through the code.